### PR TITLE
Introduce migration guide for breaking changes

### PR DESCRIPTION
**Pull request by viktormalik**
The idea is that the document will contain one section for each breaking change with:
- link to the PR introducing the change
- brief description of the change
- example of an error message
- simple guide to fix existing scripts

As an example, add entry for the recent change of the type of the `pid` and `tid` builtins.

##### Checklist

- [ ] Language changes are updated in `man/adoc/bpftrace.adoc`
- [ ] User-visible and non-trivial changes updated in `CHANGELOG.md`
- [ ] The new behaviour is covered by tests

### CODE REVIEW

File: docs/breaking.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+### `pid` and `tid` builtins return `uint32`
+
+https://github.com/bpftrace/bpftrace/pull/3441
+
+Previously, `pid` and `tid` builtins returned `uint64` so it is now possible to
+get an error when storing the builtin in a variable and overriding it with
+`uint64` later:
+```
+# bpftrace -e 'BEGIN { $x = pid; $x = uid; }'   # uid is uint_64
```
Comment (ajor) [inline docs/breaking.md:18]: `uid` might not be a very good example - it really should be a uint32 like pid and we could change that in the future too

**Review by ajor — APPROVED** [review]

File: docs/breaking.md (hunk 1 above)
Comment (viktormalik) [reply docs/breaking.md]: Good point, changed to `cgroup` which is `uint64` (returned by the BPF helper so very unlikely to change).

File: CHANGELOG.md [hunk 2]
```diff
@@ -11,6 +11,7 @@ and this project adheres to
 #### Breaking Changes
 - Return `uint32` instead of `uint64` for `pid` and `tid` builtins
   - [#3441](https://github.com/bpftrace/bpftrace/pull/3441)
+  - [Migration guide](docs/breaking.md#pid-and-tid-builtins-return-uint32)
```
Comment (jordalgo) [inline CHANGELOG.md:14]: Nit: maybe this should just go as a paragraph under the breaking changes header.

File: docs/breaking.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
+# Breaking changes (migration guide)
+
+This document lists changes introduced in bpftrace which break backwards
+compatibility in some way. Each entry should contain:
+- a link to the PR introducing the change
+- a brief description of the change
+- an example of an error message
+- a simple guide to fix existing scripts
+
+### `pid` and `tid` builtins return `uint32`
```
Comment (jordalgo) [inline docs/breaking.md:10]: We should add the versions as a `##` sub header. Maybe something like
```
## Migrating from 0.21.2 or earlier to 0.22.x or later
```

File: docs/breaking.md [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+https://github.com/bpftrace/bpftrace/pull/3441
```
Comment (jordalgo) [inline docs/breaking.md:12]: Maybe additional subheads for these to make it super clear e.g.
```
#### PR

#### Error

#### Fix
```

**Comment by jordalgo** [comment]
Yay for new migration guide! Can we also add a link from the main readme? I might also change the file name to "migration_guide" instead of "breaking" or "breaking_changes".

File: docs/breaking.md (hunk 4 above)
Comment (viktormalik) [reply docs/breaking.md]: Too much visual noise for my liking, I'd prefer not to add it.

**Comment by viktormalik** [comment]
> Yay for new migration guide! Can we also add a link from the main readme? I might also change the file name to "migration_guide" instead of "breaking" or "breaking_changes".

Sure, good idea, added.

File: CHANGELOG.md (hunk 2 above)
Comment (jordalgo) [reply CHANGELOG.md]: Oh sorry, guess I missed that.

**Review by jordalgo — APPROVED** [review]
📝

File: docs/breaking.md (hunk 4 above)
Comment (jordalgo) [reply docs/breaking.md]: Yeah, doesn't look like it needs when viewing the rendered markdown.

**Pull request merged**